src/prognostic_equations/surface_temp.jl

```diff
@@ -37,25 +36,37 @@ function surface_temp_tendency!(Yₜ, Y, p, t, slab::PrognosticSurfaceTemperatur
         Q = FT(0)
     end
 
-    # TODO: add an energy contribution from melting snow at the surface
-
     @. Yₜ.sfc.T -=
         (sfc_rad_e_flux + sfc_turb_e_flux + Q) /
         (ρ_ocean * cp_ocean * depth_ocean)
 
     if !(p.atmos.moisture_model isa DryModel)
+
+        # ENERGY (correction due to precipitation removal)
+        pet = p.conservation_check.col_integrated_precip_energy_tendency
+        earth_area = CA.horizontal_integral_at_boundary(zeros(axes(pet)) .+ 1)
+        if p.atmos.precip_model isa Microphysics0Moment
+            # correction distributed over all surface average
+            @. Yₜ.sfc.T -=
+                sum(pet) / (ρ_ocean * cp_ocean * depth_ocean) / earth_area
+        end
+
+        # @. Yₜ.sfc.T -=
+        #     p.conservation_check.col_integrated_precip_energy_tendency /
+        #     (ρ_ocean * cp_ocean * depth_ocean)
```

Comment:
What would happen if you do this? They should be equivalent? If this works I would prefer this, as it is cleaner.

Reply:
It was less stable. With `0M` the energy contribution seems to abrupt in some edge cases. I'd imagine `1M` won't have that problem, which is why I left this here. That said, I can try to run the current long runs for this being uncommented, and if they pass I can change it back.

Reply:
Maybe just increase depth_ocean to see if that helps?

Reply:
(although that's cheating) :-P

Reply:
It works! :)